aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -436,6 +458,17 @@ module AptosFramework::Token {
         };
 
         Table::add(collections, &name, collection);
+        let collection_handle =  borrow_global_mut<Collections>(account_addr);
+        Event::emit_event<CreateCollectionEvent>(
+            &mut collection_handle.create_collection_events,
+            CreateCollectionEvent {
+                creator: account_addr,
+                collection_name: *&name,
```

Comment:
Question here since I'm new to Move, ASCII::String struct is copy so technically we don't need to do *&name here right since it'll get copied by default (instead of being moved).

Reply:
I think the reason is to do explicit copy here ( here is the reference https://diem.github.io/move/variables.html). Basically, each local variable is either moved or copied. The compiler does inference based on rules if not specified Any scalar value with the copy [ability](https://diem.github.io/move/abilities.html) is given a copy.
Any reference (both mutable &mut and immutable &) is given a copy.
Except under special circumstances where it is made a move for predictable borrow checker errors.
Any other value is given a move.
This means that even though other values might be have the copy [ability](https://diem.github.io/move/abilities.html), it must be done explicitly by the programmer.
This is to prevent accidental copies of large data structures.